admin_guide/node_problem_detector.adoc

```diff
@@ -0,0 +1,426 @@
++
+[source,yaml]
+----
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: node-problem-detector
+spec:
+  selector:
+    matchLabels:
+      name: node-problem-detector
+  template:
+    metadata:
+      labels:
+        app: node-problem-detector
+    spec:
+      containers:
+      - name: node-problem-detector
+        command:
+        - /node-problem-detector
+        - --address=127.0.0.1 <1>
+        - --port=20256 <1>
+        - --logtostderr
+        - --kernel-monitor=/config/kernel-monitor.json <2>
+        image: k8s.gcr.io/node-problem-detector:v0.2
+        imagePullPolicy: Always
+        securityContext:
+          privileged: true
+        env:
+        - name: NODE_NAME
+          valueFrom:
+            fieldRef:
+              fieldPath: spec.nodeName
+        volumeMounts:
+        - name: log
+          mountPath: /var/log
+          readOnly: true
+        - name: kmsg
+          mountPath: /dev/kmsg
+          readOnly: true
+        # Make sure node problem detector is in the same timezone
+        # with the host.
+        - name: localtime
+          mountPath: /etc/localtime
+          readOnly: true
+        - name: config
+          mountPath: /config
+          readOnly: true
+      volumes:
+      - name: log
+        # Config `log` to your system log directory
+        hostPath:
+          path: /var/log/ <3>
+      - name: kmsg
+        hostPath:
+          path: /dev/kmsg
+      - name: localtime
+        hostPath:
+          path: /etc/localtime
+      - name: config
+        configMap:
+          name: node-problem-detector-config <4>
+----
++
+<1> Node problem detector binding address and port.
+<2> List of paths to link:https://github.com/kubernetes/node-problem-detector/blob/master/config/kernel-monitor.json[system log monitor configuration files], 
+comma separated. Node problem detector will start a separate log monitor for each configuration.
+<3> Path to your system log directory.
+<4> Configuration map used by the node problem detector. You can xref:admin-guide-node-problem-detector-overwrite[use your own configuration file], as needed.
+
+. Edit the file for your environment. 
+
+. Save the file as any name, such as *_node-problem-detector.yaml_*.
+
+. Run the following command to start node problem detector: 
++
+[source,bash]
+----
+oc create -f <your_yaml_file>.yaml
+----
+
+[[admin-guide-node-problem-detector-customize]]
+== Configuring Custom Logging and Monitoring Tools
+
+You can use xref:admin-guide-node-problem-detector-fbk[file-based kernel logging], or 
+xref:admin-guide-node-problem-detector-monitor[custom monitoring tools]. 
+
+To enable plugins, edit the node problem detector daemonset:
+
+. Edit the Node Problem Detector configuration map with a text editor.
+[source,bash]
++
+----
+oc edit ds -n <project> configmap <node-problem-detector-name>
+----
++
+For example:
++
+----
+oc edit ds -n openshift-node-problem-detector configmap node-problem-detector
```

Comment:
Sorry if I told you wrong before. This one should not have `ds` in it. This command is to edit the `configmap` (which comes later in the command).

If you prefer to have the `-n <project>` later in the command, then this should read
```
oc edit configmap -n openshift-node-problem-detector node-problem-detector
```
or if you prefer the ns first (like the previous version), it would look like:
```
oc edit -n openshift-node-problem-detector configmap node-problem-detector
```